### PR TITLE
introduces a few convenient functions for querying netlink subsystem

### PR DESCRIPTION
**Pull request by p0lyn0mial**
`netlink` can be used for obtaining information about network sockets of various address families from the Linux kernel.

more information can be found at https://man7.org/linux/man-pages/man7/sock_diag.7.html

### CODE REVIEW

File: pkg/network/networkutils/netlink.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+	// remoteAddress holds an IP address and the port of the remote process
+	remoteAddress string
+
+	// timer - see timerFieldToHumanReadable for possible values
+	timer string
+
+	// retransmissions - for timer values 1, 2, and 4, this field contains
+	// the number of retransmits. For other timer values, this field is set to 0.
+	retransmissions int
```
Comment (p0lyn0mial) [inline pkg/network/networkutils/netlink.go:45]: should be renamed to `timerRetransmissions`

File: pkg/network/networkutils/netlink.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+// This method uses netlink subsystem for obtaining information
+// about sockets of various address families from the linux kernel.
+//
+// in the future we could:
+//  use InetDiagReqV2 to specify an address family (IPv4 or IPv6) and a protocol (TCP, UDP)
+//  ask the kernel to only return sockets for specific port or address (filtering)
+//
+// Note:
+//  depending on the platform this function is run on, it might require setting custom permissions.
+func GetNetworkSocketsDiagnosticMsg() ([]*SocketDiagMsg, error) {
```
Comment (sttts) [inline pkg/network/networkutils/netlink.go:91]: "Get" sounds like a no-cost operation. This is more like "Collect" or "Gather".

File: pkg/network/networkutils/netlink.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
+// +build linux
```
Comment (sttts) [inline pkg/network/networkutils/netlink.go:1]: we need a no-op replacement for other platforms

Comment (p0lyn0mial) [reply pkg/network/networkutils/netlink.go:1]: for each method?

Would it be okay to add a replacement when composing the final app ?

For example, the final app could have two files `app.go and app_others.go`.
Then `app_others.go` could simply print that running it is not supported on that platform.

Comment (sttts) [reply pkg/network/networkutils/netlink.go:1]: yes, could work too

Comment (deads2k) [reply pkg/network/networkutils/netlink.go:1]: > Would it be okay to add a replacement when composing the final app ?

Seems like the library shouldn't just flat out fail to compile on other platforms.

**Comment by p0lyn0mial** [comment]
are we actually still interested in it?

**Pull request closed without merging**